compiler-tests/src/test/data/box/cycles/AssistedFactoryCycle.kt

```diff
@@ -0,0 +1,33 @@
+@Inject
+class Foo(
+  @Assisted nested: Boolean,
+  factoryProvider: Provider<Factory>,
+  factoryLazy: Lazy<Factory>,
```

Comment:
Dagger doesn't appear to support this case in my testing, are you seeing otherwise?

> error: Dagger does not support injecting Lazy<T>, Producer<T>, or Produced<T> when T is an @AssistedFactory-annotated type such as ...Foo.Factory

Reply:
Ahh, you're totally right! I tested only the Provider, and thought Lazy would work too. Will update the test case 🙏